gaia-header.js

```diff
@@ -364,11 +371,51 @@ module.exports = component.register('gaia-header', {
   observerStop: function() {
     if (!this.observing) { return; }
     this.observer.disconnect();
+
     this.observing = false;
     debug('observer stopped');
   },
 
   /**
+   * Handle DOM events.
+   * @param {Event} The DOM Event that's being handled.
+   *
+   * @private
+   */
+  handleEvent: function(e) {
+    debug('received event', e.type, e);
+    switch(e.type) {
+      case 'resize':
+        this.onResize();
+        break;
+    }
+  },
+
+  /**
+   * Handle 'resize' events.
+   *
+   * @private
+   */
+  onResize: function() {
+    debug('onResize', this._resizeThrottlingId);
+
+    if (this._resizeThrottlingId !== null) {
+      return;
+    }
+
+    if (this.notFlush) {
```

Comment:
I did not want to run it in that case for 2 reasons:
* we don't use innerWidth but clientWidth... which, you got it, triggers a reflow (although we're after requestAnimationFrame, so maybe we don't have a reflow in this case)
* you can also have a fixed-size header, and in that case I thought that I didn't want to get clientWidth just to bail out later when we find that styleId is the same. I thought that in the "notFlush" case we could let the user run runFontFit manually if necessary.

But I don't have a strong feeling either. If you really think this should run, I can do it.